functions/index.js

```diff
@@ -1,174 +1,224 @@
+        title += ' rescheduled the appointment';
+        const msg = 'Appointment moved to ' + newValue.date.toDate();
+        const notificationContent = {
+            notification: {
+              title: title,
+              body: msg,
+              sound: 'default'
+            },
+            data: {
+              client_id: clientDocID,
+              appointment_id: appointmentId,
+              category: 'NotificationCategory.appointment',
+            }
+        };
+        console.log('Send notification about the rescheduled appointment')
+        if (changedByClient) {
+            const adminTokens = await getAdminTokens();
+            return admin.messaging().sendToDevice(adminTokens, notificationContent);
+        } else {
+            const clientTokens = await getClientTokens(clientDocID);
+            if(clientTokens) {
+                console.log('Sending notification to ', newValue.clientName)
+                return admin.messaging().sendToDevice(clientTokens, notificationContent);
+            }
+        }
     }
-    if (filePath.startsWith('clients/')) {
-        return clientsImagesHandler(object);
+    if (newValue.status != previousValue.status) {
+        if(newValue.status === 'AppointmentStatus.confirmed') {
+            console.log('Send notification about confirmed appointment')
+            const title = 'Aleen confirmed your appointment';
+            const msg = 'We are expecting you on ' + newValue.date.toDate();
+            const notificationContent = {
+               notification: {
+                 title: title,
+                 body: msg,
+                 sound: 'default'
+               },
+               data: {
+                 client_id: clientDocID,
+                 appointment_id: appointmentId,
+                 category: 'NotificationCategory.appointment',
+               }
+           };
+           const clientTokens = await getClientTokens(clientDocID);
+           if(clientTokens) {
+               console.log('Sending notification to ', newValue.clientName)
+               return admin.messaging().sendToDevice(clientTokens, notificationContent);
+           }
+        }
+        if(newValue.status === 'AppointmentStatus.declined') {
+            console.log('Send notification about declined appointment')
+            const title = 'Failed to schedule your appointment';
+            const msg = 'Please contact Aleen to schedule the appointment';
+            const notificationContent = {
+               notification: {
+                 title: title,
+                 body: msg,
+                 sound: 'default'
+               },
+               data: {
+                 client_id: clientDocID,
+                 appointment_id: appointmentId,
+                 category: 'NotificationCategory.appointment',
+               }
+           };
+           const clientTokens = await getClientTokens(clientDocID);
+           if(clientTokens) {
```

Comment:
what if client is the one who cancelled the appointment ?
we should send the msg to admin

Reply:
sorry, forget this, Its declined status